sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1175,15 +1187,69 @@ class CompactBlockProcessor internal constructor(
                     .map {
                         SubtreeRoot.new(it, network)
                     }.let {
-                        result =
-                            if (it.isEmpty()) {
-                                GetSubtreeRootsResult.Linear
+                        saplingSubtreeRootList = it
+                    }
+            }
+
+            retryUpToAndContinue(GET_SUBTREE_ROOTS_RETRIES) {
+                downloader.getSubtreeRoots(
+                    orchardStartIndex,
+                    maxEntries = UInt.MIN_VALUE,
+                    shieldedProtocol = ShieldedProtocolEnum.ORCHARD
+                ).onEach { response ->
+                    when (response) {
+                        is Response.Success -> {
+                            Twig.verbose {
+                                "Orchard SubtreeRoot fetched successfully: its completingHeight is: ${response.result
+                                    .completingBlockHeight}"
+                            }
+                        }
+                        is Response.Failure -> {
+                            val error =
+                                LightWalletException.GetSubtreeRootsException(
+                                    response.code,
+                                    response.description,
+                                    response.toThrowable()
+                                )
+                            if (response is Response.Failure.Server.Unavailable) {
+                                Twig.error {
+                                    "Fetching Orchard SubtreeRoot failed due to server communication problem with " +
+                                        "failure: ${response.toThrowable()}"
+                                }
+                                result = GetSubtreeRootsResult.FailureConnection
                             } else {
-                                GetSubtreeRootsResult.SpendBeforeSync(startIndex, it)
+                                Twig.error {
+                                    "Fetching Orchard SubtreeRoot failed with failure: ${response.toThrowable()}"
+                                }
+                                result = GetSubtreeRootsResult.OtherFailure(error)
                             }
+                            traceScope.end()
+                            throw error
+                        }
+                    }
+                }
+                    .filterIsInstance<Response.Success<SubtreeRootUnsafe>>()
+                    .map { response ->
+                        response.result
+                    }
+                    .toList()
+                    .map {
+                        SubtreeRoot.new(it, network)
+                    }.let {
+                        orchardSubtreeRootList = it
                     }
             }
 
+            if (!saplingSubtreeRootList.isEmpty()) {
```

Comment:
```suggestion
            if (!(saplingSubtreeRootList.isEmpty() && orchardSubtreeRootList.isEmpty())) {
```
Or explain why we don't need to check the Orchard roots in a comment.

Reply:
I've just added the `orchardSubtreeRootList` in the result check with [8df0b13](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1411/commits/8df0b139d77fa42dd4b417fdb95726f0a364d536). BUT I'd rather check this with @str4d if omitting it wasn't our intention.

Reply:
The original code checked if Sapling subtree roots were empty to detect a lightwalletd server that couldn't be used for SbS. I kept it that way because if the Orchard subtree roots are empty, that's technically fine (as Orchard activates after Sapling, so on a network that doesn't have NU5 activated this would behave correctly), whereas the Sapling subtree roots are empty, we cannot do SbS at all.

Reply:
Thank you for the explanation. The `orchardSubtreeRootList` check reverted and commented out inline with [64484c5](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1411/commits/64484c580c7983589514788ed474e1dfd3a49d8a)